src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateCopy = {
+          ...stateCopy,
+          ...action.extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
```

Comment:
You can use a destruction action
```suggestion
  for (const action of actions) {
    const { type, extraData, keysToRemove } = action;
    switch (type) {
      case 'addProperties':
        stateCopy = {
          ...stateCopy,
          ...extraData,
        };
        break;

      case 'removeProperties':
        for (const key of keysToRemove) {
```